OpenDreamRuntime/Objects/DreamObjectTree.cs

```diff
@@ -70,6 +78,32 @@ public class TreeEntry {
             }
         }
 
+        /// <summary>
+        /// Returns a list of the TreeEntry's Types sorted in DFS traversal order
+        /// </summary>
+        /// <returns>IEnumerable of sorted TreeEntries</returns>
+        /// <seealso cref="DreamObjectDefinition.IsSubtypeOf">IsDescendantOf</seealso>
+        public IEnumerable<TreeEntry> GetDFSSortedTypeList() {
+            List<TreeEntry> dfs_sorted_types = new();
+            Stack<TreeEntry> dfs_sort_stack = new();
+
+            dfs_sort_stack.Push(Types[0]); // Always the root ('/')
```

Comment:
Add a `public TreeEntry RootType { get => Types[0]; }` property and use that here instead.